timemachine/fe/mcgregor.py

```diff
@@ -426,6 +465,7 @@ def recursion(
     enforce_core_core,
     max_connected_components: Optional[int],
     min_connected_component_size: int,
+    disallow_chiral_conversion: bool,
```

Comment:
Not sure if this is the right name here - since we also disallow sp3 chiral conversions at the atom-mapping level. And maybe this should be generalized into `leaf_filter_fxn`? i.e. a certificate we check at the leaf nodes?

Reply:
Was thinking the same thing -- that would also let us move the single-topology-specific code out of `mcgregor`

Reply:
Refactored the interface to expose a general `leaf_filter_fxn` argument to `mcs` in https://github.com/proteneer/timemachine/pull/1356/commits/214d8d03821dc319e624957c0c2d74bea3d35b4b